### ISSUE TITLE
spacy 2.0 seems to cause problems

### ISSUE THREAD

**Issue by yayitswei** [HazyResearch/snorkel#830]
```
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-2-dd528794d4e5> in <module>()
----> 1 from snorkel.parser import TSVDocPreprocessor
      2 
      3 doc_preprocessor = TSVDocPreprocessor('data/articles.tsv', max_docs=n_docs)

/Users/wei/code/snorkel/snorkel/parser/__init__.py in <module>()
      3 from .doc_preprocessors import *
      4 from .parser import *
----> 5 from .spacy_parser import *
      6 from .rule_parser import *

/Users/wei/code/snorkel/snorkel/parser/spacy_parser.py in <module>()
      9     from spacy.deprecated import resolve_model_name
     10 except:
---> 11     raise Exception("spaCy not installed. Use `pip install spacy`.")
     12 
     13 

Exception: spaCy not installed. Use `pip install spacy`.
```

**Comment by ajratner** [comment]
So you have `spacy` installed and then this happened when you upgraded?  Can you give a bit more context?  thanks!

**Comment by yayitswei** [comment]
Yup, this happened with spacy 2.0 and when I downgraded to spacy 1.10.0 I didn't get this message anymore.

**Comment by piordanov** [comment]
Spacy v2.0 has a lot of backwards incompatibilities listed [here](https://spacy.io/usage/v2). It's likely one of these must be impacting snorkel

**Comment by DarkDataAnalytics** [comment]
I have the same issue. Even when uninstalling space 2.0.3 and then installing space 1.10.0.
Yayitswei how did you downgraded spacy ? Did you do a pip uninstall ?

**Comment by yayitswei** [comment]
I added this to python-package-requirement.txt:

```
spacy==1.10.0
```

**Comment by yayitswei** [comment]
I added this to `python-package-requirement.txt`:

```
spacy==1.10.0
```

**Comment by ajratner** [comment]
Hi @yayitswei if you want to submit that change as a PR that would be great :) . Otherwise we'll add shortly

**Comment by DarkDataAnalytics** [comment]
Thanks for the answer but it is still not working for me.

**Comment by DarkDataAnalytics** [comment]
A bit of context here.
I use Ubuntu 16.04 on an EC2 instance with Python 2.7. The OS is up to date.
When I install the libraries with pip I do :
sudo -H pip install --requirement python-package-requirement.txt
I do it after doing : source avtivate py2Env

**Comment by fsonntag** [comment]
Yeah, it's because of the `from spacy.deprecated import resolve_model_name`, because this functionality is gone. You can remove this and replace
``` 
        model_name = resolve_model_name(name)
        model_path = data_path / model_name

```
with 
```
        model_path = data_path / name

```
Works fine.

**Comment by ajratner** [comment]
Should be fixed by PR #834 re-open if not!

**Comment by arturomp** [comment]
Using spacy 2.0.5, snorkel 0.6.2 (acc. to `snorkel/snorkel/__init__.py`) or 0.6.0 (acc. to `snorkel/docs/conf.py`). Ran a slightly modified version of the intro tutorial. Had to change `spacy_parser.py` due to the error

    ---------------------------------------------------------------------------
    KeyError                                  Traceback (most recent call last)
    <ipython-input-3-42840d262509> in <module>()
         36 #             yield Sentence(**parts)
         37 
    ---> 38 corpus_parser = CorpusParser(parser=Spacy(lang='en_core_web_lg'))
         39 get_ipython().magic(u'time corpus_parser.apply(doc_preprocessor, count=n_docs)')

    /Users/ampawake/Documents/awake/code/rudi.nlp/ietests/snorkel/snorkel/parser/spacy_parser.py in __init__(self, annotators, lang, num_threads, verbose)
         59         self.pipeline = []
         60         for proc in annotators:
    ---> 61             self.pipeline += [self.model.__dict__[proc]]
         62 
         63     @staticmethod

    KeyError: u'tagger'

Replaced:

    self.pipeline = []
    for proc in annotators:
        self.pipeline += [self.model.__dict__[proc]]

for

    self.pipeline = [proc for _, proc in self.model.__dict__['pipeline']]

If using spacy >= 2, you won't need the values of `annotators` anymore in `Spacy.__init__()` in `spacy_parser`. But just in case, their values now seem to be `'tagger', 'parser', 'ner'` instead of `'tagger', 'parser', 'entity'`.
